### ISSUE TITLE
Feature request: log search / filter

### ISSUE THREAD

**Issue by fabriziopandini** [vmware-tanzu/octant#213]
As a developer, I look at logs all the time... it would be great to:
- be able to search for a specific text/regex inside logs
- filter logs lines for a specific text/regex (grep like)

It will be great also to have some text automatically highlighted like e.g. error, panic etc.

**Comment by GuessWhoSamFoo** [comment]
@fabriziopandini Could you clarify a bit on the difference between the two bullet points? We're planning to break this apart into three different feature requests.

**Comment by fabriziopandini** [comment]
@GuessWhoSamFoo sure!

> be able to search for a specific text/regex inside logs

Users to specify a regex/a search string --> octant should highlight matches in the log and provide some buttons to quickly move from one match to the following/previous

> filter logs lines for a specific text/regex (grep like)

Users to specify a regex/a search string --> octant should filter out (hide) all the logs lines without a match and highlight matches on the remaining lines

TL;DR; with the first option you still see the full logs, while in the second option the logs are "reduced" to only the lines with matches

**Comment by mklanjsek** [comment]
I am thinking about implementing this through a very simple UX, something like this:

**Comment by wwitzel3** [comment]
An example of a filter for logs I really like was the one in GitHub Actions

https://github.com/vmware-tanzu/octant/runs/439734021?check_suite_focus=true

Withe the `<` and `>` for forward and back and the `...` to hide the timestamp / raw options.

**Comment by mklanjsek** [comment]
@wwitzel3 pretty slick, I like that.

**Comment by mklanjsek** [comment]
Status: very close with log filter/search implementation, having some issues with moving between selections

**Comment by fabriziopandini** [comment]
Looking forward to test this! Thanks!
